pkg/admission/cert/writer/certwriter.go

```diff
@@ -141,18 +151,36 @@ type certReadWriter interface {
 	overwrite(webhookName string) (*generator.Artifacts, error)
 }
 
-func validCert(certs *generator.Artifacts) bool {
-	// TODO:
-	// 1) validate the key and the cert are valid pair e.g. call crypto/tls.X509KeyPair()
-	// 2) validate the cert with the CA cert
-	// 3) validate the cert is for a certain DNSName
-	// e.g.
-	// c, err := tls.X509KeyPair(cert, key)
-	// err := c.Verify(options)
+func validCert(certs *generator.Artifacts, dnsName string) bool {
 	if certs == nil {
 		return false
 	}
+
+	// Verify key and cert are valid pair
 	_, err := tls.X509KeyPair(certs.Cert, certs.Key)
+	if err != nil {
+		return false
+	}
+
+	// Verify cert is good for desired DNS name and signed by CA and will be valid for desired period of time.
+	pool := x509.NewCertPool()
+	if !pool.AppendCertsFromPEM(certs.CACert) {
+		return false
+	}
+	block, _ := pem.Decode([]byte(certs.Cert))
+	if block == nil {
+		return false
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return false
+	}
+	ops := x509.VerifyOptions{
+		DNSName:     dnsName,
+		Roots:       pool,
+		CurrentTime: time.Now().Add(sixMonths),
```

Comment:
why 6 months? also it's more like `time.Now().AddDate(0, 1,  0)` [doc](https://golang.org/pkg/time/#Time.AddDate)

Reply:
Good point!
Updated